web/src/features/prompts/server/prompt-router.ts

```diff
@@ -497,3 +543,69 @@ export const createPrompt = async ({
 
   return createdPrompt;
 };
+
+const generatePromptQuery = (
+  select: Prisma.Sql,
+  projectId: string,
+  filterCondition: Prisma.Sql,
+  orderCondition: Prisma.Sql,
+  limit: number,
+  page: number,
+) => {
+  return Prisma.sql`
+  SELECT
+   ${select}
+   FROM prompts p
+   WHERE (name, version) IN (
+    SELECT name, MAX(version)
+     FROM prompts
+     WHERE "project_id" = ${projectId}
+          GROUP BY name
+        )
+    AND "project_id" = ${projectId}
+  ${filterCondition}
+  ${orderCondition}
+  LIMIT ${limit} OFFSET ${page * limit};
+`;
+};
+
+function filterPromptsByCount(
```

Comment:
either do this via sql or not at all, this does not work anymore when we add pagination to this table

Reply:
let's remove the filter and order of count alltogether